lib/dolly/document.rb

```diff
@@ -149,6 +151,20 @@ def self.property *ary
       end
     end
 
+    class << self
+      def mango_scope scope_name, scope
+        self.mango_scopes ||= {}
+        name = scope_name.to_sym
+        self.mango_scopes[name] = lambda { |query_object, args| Dolly::Scope.new(query_object, scope, args)}
+
+        (class << self; self end).instance_eval do
```

Comment:
what is this witchcraft!?  

interesting, can you elaborate on what is happening?